spec/filters/git_last_checkout_filter_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+RSpec.describe Airbrake::Filters::GitLastCheckoutFilter do
+  subject { described_class.new(Logger.new(STDOUT), '.') }
+
+  let(:notice) do
+    Airbrake::Notice.new(Airbrake::Config.new, AirbrakeTestError.new)
+  end
+
+  context "when context/lastCheckout is defined" do
+    it "doesn't attach anything to context/lastCheckout" do
+      notice[:context][:lastCheckout] = '123'
+      subject.call(notice)
+      expect(notice[:context][:lastCheckout]).to eq('123')
+    end
+  end
+
+  context "when .git directory doesn't exist" do
+    subject { described_class.new(Logger.new(STDOUT), 'root/dir') }
+
+    it "doesn't attach anything to context/lastCheckout" do
+      subject.call(notice)
+      expect(notice[:context][:lastCheckout]).to be_nil
+    end
+  end
+
+  context "when .git directory exists" do
+    before { subject.call(notice) }
+
+    %i[username email revision time].each do |param|
+      it "attaches last checkouted #{param}" do
+        expect(notice[:context][:lastCheckout][param]).not_to be_empty
```

Comment:
It would be nice to see at least name and email here...

Reply:
Unless we mock it, it will keep changing depending on who committed.